katdal/test/test_chunkstore_s3.py

```diff
@@ -0,0 +1,103 @@
+################################################################################
+# Copyright (c) 2011-2016, National Research Foundation (Square Kilometre Array)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Tests for :py:mod:`katdal.chunkstore_s3`."""
+
+import tempfile
+import shutil
+import subprocess
+import time
+
+import numpy as np
+from numpy.testing import assert_array_equal
+from nose import SkipTest
+from nose.tools import assert_raises
+import requests
+
+import katdal.chunkstore_s3
+from katdal.chunkstore_s3 import S3ChunkStore
+
+
+class TestS3ChunkStore(object):
+    def setup(self):
+        """Start Fake S3 service running on temp dir, and ChunkStore on that."""
+        self.tempdir = tempfile.mkdtemp()
+        host = 'localhost'
+        # Choose a random port and hope it's available
+        port = np.random.randint(10000, 60000)
+        url = "http://%s:%d" % (host, port)
+        try:
+            self.fakes3 = subprocess.Popen(['fakes3', 'server', '-H', host,
+                                            '-p', str(port), '-r', self.tempdir])
```

Comment:
Also pass `-a 127.0.0.1`, otherwise it listens on all interfaces, which is a security risk (I think `-H` just sets the advertised hostname in responses).

Reply:
Done.